src/components/Filters/StatefulFilters.tsx

```diff
@@ -216,7 +237,7 @@ export class StatefulFilters extends React.Component<StatefulFiltersProps, State
   renderInput() {
     const { currentFilterType, currentValue } = this.state;
 
-    if (!currentFilterType) {
+    if (!currentFilterType || currentFilterType.filterType === FilterTypes.opLabel) {
```

Comment:
Is there something blocking for *not* having this "op labels" as a standard filter? It sounds a bit odd and forces to make a lot of special case handling like on this line

Reply:
For me it makes not sense has a filter if there is not labels selected. On the other hand we are going to apply the "Or" operation if the user not set "And" so...Should we add a filter when the list is loaded? When the user set the first label ? ...

For these reasons I thought about have this behavior. What do you think?

Reply:
I mean, I'm questioning the idea to have `opLabel` as a `FilterTypes`. It isn't a filter, it's something different.

Reply:
Understood...I added it to be able to pass the option to the others list, we can handle this in another way

> I had to add this like a new Filter to be able to move this param to other lists

On the other hand we'll need to think another way to pass it to the filter method
https://github.com/kiali/kiali-ui/pull/1780/files#diff-f1cc29563c6473251f2ce0e4043b27e6R318

Other solution is change the "LabelFilter" to let have more params like this operation

Reply:
I'm trying to see how we could do differently, as having `opLabel` as a `FilterTypes` looks more like a hack.

Perhaps, instead of having that new `OpLabelFilter` object in `Filters.ts`, we would change how we use `ActiveFilter`. Something like that (in Filters.ts):

```typescript
// This already exists
export interface ActiveFilter {
  category: string;
  value: string;
}

// This is new
export type LabelOperation = 'and' | 'or';
export interface ActiveFiltersInfo {
  filters: ActiveFilter[];
  op: LabelOperation;
}
```

And then, in many place where we use `ActiveFilter[]`, we would use `ActiveFiltersInfo`.
For instance in `StatefulFilters` state we would have `activeFilters: ActiveFiltersInfo;`

That's a bigger refactoring.

Reply:
yes I know,....but I expect more changes/filters in this line so maybe we must do it now or later ^^

Reply:
After your refactoring, I think this `opLabel` in `FilterTypes` isn't used anymore, and can be safely removed, right?